apps/api/src/app/auth/usecases/register/user-register.usecase.ts

```diff
@@ -29,11 +30,14 @@ export class UserRegister {
     if (existingUser) throw new ApiException('User already exists');
 
     const passwordHash = await bcrypt.hash(command.password, 10);
+    const intercomSecretKey = process.env.INTERCOM_IDENTITY_VERIFICATION_SECRET_KEY as string;
+    const userHashForIntercom = createHmac('sha256', intercomSecretKey).update(email).digest('hex');
```

Comment:
Let's extract this to a helper function so we can reuse this logic in a single place